### PR TITLE
Added compatibility for NoSQL DB

### PR DESCRIPTION
**Pull request by kjain0073**
Hi Vasu, The boilerplate is now compatible with NoSQL as well as SQL Databases. Kindly look into it and merge the changes

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,20 +21,34 @@ func main() {
 	defer recoverPanic()
 	ctx := context.Background()
 	//Initialize the config
-	if err := config.InitConfig(); err != nil {
+
+	if err := config.InitConfig("mongodb"); err != nil {
```
Comment (jinagamvasubabu) [inline main.go]: dont modify the `InitConfig`  method, instead set default value to  true for either mongo/postgres in config file

File: main.go [hunk 2]
```diff
@@ -21,20 +21,34 @@ func main() {
...
+	//get config to check DB Type
+	cfg := config.GetConfig()
+	var bookRepository repository.BookRepository
+
+	if cfg.DBType == "postgres" {
+		DB, err := db.InitPostgresDatabase()
+		if err != nil {
+			log.Errorf("Error:%s", err.Error())
+		}
+		bookRepository = repository.PostgresNewBookRepository(ctx, DB)
```
Comment (jinagamvasubabu) [inline main.go]: rename the method to `NewPostgresBookRepository`

File: main.go [hunk 3]
```diff
@@ -21,20 +21,34 @@ func main() {
...
+	} else {
+		DB, err := db.InitMongoDatabase()
+		if err != nil {
+			log.Errorf("Error:%s", err.Error())
+		}
+		bookRepository = repository.MongoNewBookRepository(ctx, DB, cfg)
```
Comment (jinagamvasubabu) [inline main.go]: same here

File: repository/book_postgresRepository.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+func (b postgresBookRepository) GetAllBooks(ctx context.Context) ([]model.Book, error) {
+	books := []model.Book{}
+	if result := b.db.Find(&books); result.Error != nil {
+		Logger.Errorf("Error while finding books = %s", result.Error.Error())
+		return books, result.Error
+	}
+	return books, nil
+}
+
+//MONGODB
```
Comment (jinagamvasubabu) [inline repository/book_postgresRepository.go]: remove the commented code

**Comment by jinagamvasubabu** [comment]
@kjain0073 added the review comments for the same also please add the docker for mongodb as well if you can

**Comment by slice-vasuj** [comment]
LGTM, Merging it

**Pull request merged**